src/shogun/machine/gp/SingleFITCLaplacianBase.cpp

```diff
@@ -45,73 +45,30 @@
 using namespace shogun;
 using namespace Eigen;
 
-CSingleFITCLaplacianBase::CSingleFITCLaplacianBase() : CFITCInferenceBase()
+CSingleFITCLaplacianBase::CSingleFITCLaplacianBase() : CSingleSparseInferenceBase()
 {
 	init();
 }
 
 CSingleFITCLaplacianBase::CSingleFITCLaplacianBase(CKernel* kern, CFeatures* feat,
 		CMeanFunction* m, CLabels* lab, CLikelihoodModel* mod, CFeatures* lat)
-		: CFITCInferenceBase(kern, feat, m, lab, mod, lat)
+		: CSingleSparseInferenceBase(kern, feat, m, lab, mod, lat)
 {
 	init();
-	check_fully_FITC();
 }
 
 void CSingleFITCLaplacianBase::init()
 {
-	m_fully_FITC=false;
-	m_lock=new CLock();
 	SG_ADD(&m_al, "al", "alpha", MS_NOT_AVAILABLE);
 	SG_ADD(&m_t, "t", "noise", MS_NOT_AVAILABLE);
 	SG_ADD(&m_B, "B", "B", MS_NOT_AVAILABLE);
 	SG_ADD(&m_w, "w", "B*al", MS_NOT_AVAILABLE);
 	SG_ADD(&m_Rvdd, "Rvdd", "Rvdd", MS_NOT_AVAILABLE);
 	SG_ADD(&m_V, "V", "V", MS_NOT_AVAILABLE);
-	SG_ADD(&m_fully_FITC, "fully_FITC",
-		"whether the kernel support fitc inference", MS_NOT_AVAILABLE);
-
```

Comment:
Is all this deleted code just refactoring?
Again, it would be good to decompose huge patches like this one a bit. I cannot really see whether this stuff is now deleted or just moved as changes are hidden among the many others